kvx/src/implementations/disk.rs

```diff
@@ -348,63 +346,33 @@ impl PathBufExt for PathBuf {
 
 #[derive(Debug)]
 struct FileLock {
-    file: File,
-    lock_path: PathBuf,
+    lock: fd_lock::RwLock<File>,
 }
 
 impl FileLock {
-    const POLL_LOCK_INTERVAL: Duration = Duration::from_millis(10);
-
-    pub fn lock(path: impl AsRef<Path>) -> Result<Self> {
+    fn create(path: impl AsRef<Path>) -> Result<Self> {
         let path = path.as_ref();
 
-        if !path.try_exists().unwrap_or_default() {
-            fs::create_dir_all(path)?;
-        }
-
         let lock_path = path.join(LOCK_FILE_NAME);
 
-        let file = loop {
-            let file = OpenOptions::new()
-                .create_new(true)
-                .read(true)
-                .write(true)
-                .open(&lock_path);
-
-            match file {
-                Ok(file) => break file,
-                _ => thread::sleep(Self::POLL_LOCK_INTERVAL),
-            };
+        let lock_file = if lock_path.exists() {
```

Comment:
This is probably better done by trying to open the file and do all the other thing if you get an `io::ErrorKind::NotFound` error – this avoids a race condition if the file disappears between your check and being opened.

Reply:
Changed to this, which I think should avoid  race conditions:

```
let lock_file = OpenOptions::new()
            .create(true)
            .read(true)
            .write(true)
            .open(lock_path)?;
```